src/apps/authentication/api/auth.py

```diff
@@ -175,6 +183,10 @@ async def verify_mfa_totp(
             new_attempt_count = await mfa_service.increment_failed_totp_attempts(mfa_session_id)
             global_attempt_count = await mfa_service.increment_global_failed_attempts(user.id)
 
+            # Calculate remaining attempts
+            session_remaining = settings.redis.mfa_max_attempts - (new_attempt_count or 0)
+            global_remaining = settings.redis.mfa_global_lockout_attempts - global_attempt_count
```

Comment:
Is there a compelling reason to calculate `session_remaining` and `global_remaining` in-line instead of using `await mfa_service.get_remaining_session_attempts(mfa_session_id)` and `await mfa_service.get_remaining_global_attempts(user.id)`?

I would prefer relying on the `mfa_service` methods to make sure we calculate `session_remaining` and `global_remaining` in the same way everywhere.